.github/workflows/gradle-build.yml

```diff
@@ -72,7 +72,7 @@ jobs:
         with:
           workflow: preview-and-release.yml
           workflow_conclusion: success
-          branch: main
+          branch: dev
```

Comment:
I'm not sure this will work. The branch no longer exists. Yes?

Reply:
We deleted the branch as part of branching alignment. 
The historic workflow runs are still present with assets though.
@Ndiritu I don't think we need to make that change. This check is optional

Reply:
It's optional but the "Git release" workflow which actually uploads build artifacts to the release created by release-please fails because it's looking for a previous successful build workflow run. The build workflow is unsuccessful because one step fails (compare packages)

Reply:
I think there might be a misunderstanding here, let me try to bring additional context.

This job (compare-packages) is a CI job, that's not enforced as a gate check. Yes it uses the same taks/branching mechanism as the git release job, but it's orthogonal. And it's ok to have it fail the first time because the upload artifact task it depends on, hasn't run on that branch before. It should resolve itself over time.

[That job](https://github.com/microsoftgraph/msgraph-sdk-java/blob/62019c4eab96c84eede8bb483193bda203f27588/.github/workflows/git-release.yml#L10) (git release) depends on the same upload to successfully create the release, and has a dependency on a branch. Since everything now triggers on tags, I suggest we merge that workflow with the one in the next paragraph. (we'd be able to use the standard download artifact task, no more dependency on a specific branch, etc)

[This other job](https://github.com/microsoftgraph/msgraph-sdk-java/blob/62019c4eab96c84eede8bb483193bda203f27588/.github/workflows/preview-and-release.yml#L53) (maven release) is the one uploading the artefact both previous jobs depend on. Again, I'm suggesting we bring the git release job as a successor to the maven release one, simplifying everything in the process.

Reply:
This makes sense @baywet. I mis-understood where the artifacts were being pulled from. Updated the PR description & title